Define minimum version requirement for Scipy

While testing the minimum version requirements for astroML I run into the issue that for astropy, we don't have a minimum Scipy required, e.g this is with astropy 3.0 and scipy 0.16. Even there it seems that at least scipy 0.18 is required.

To avoid this and similar issues in the future, I suggest that we declare minimum dependency for all dependencies, and have a job in CI that tests those versions.

```
Package versions: 
Numpy: 1.13.0
Scipy: 0.16.0
Matplotlib: 3.0.0
Cython: not available
Astropy: 3.0
scikit-learn: 0.18
astroML_addons: not available
```
```
____________________________ ERROR collecting astroML/time_series/tests/test_periodogram.py ____________________________
astroML/time_series/periodogram.py:6: in <module>
    from astropy.timeseries import LombScargle
E   ImportError: No module named 'astropy.timeseries'

During handling of the above exception, another exception occurred:
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
astroML/time_series/__init__.py:1: in <module>
    from .ACF import ACF_scargle, ACF_EK
astroML/time_series/ACF.py:6: in <module>
    from .periodogram import lomb_scargle
astroML/time_series/periodogram.py:8: in <module>
    from astropy.stats import LombScargle
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/stats/__init__.py:18: in <module>
    from .circstats import *
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/stats/circstats.py:14: in <module>
    from astropy.units import Quantity
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/units/__init__.py:14: in <module>
    from .quantity import *
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/units/quantity.py:29: in <module>
    from .quantity_helper import (converters_and_unit, can_have_arbitrary_unit,
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/units/quantity_helper.py:457: in <module>
    sps.erf, sps.gamma, sps.loggamma, sps.gammasgn,
E   AttributeError: module 'scipy.special' has no attribute 'loggamma'
```

--- a/astropy/__init__.py
+++ b/astropy/__init__.py
@@ -14,6 +14,7 @@ from .version import version as __version__
 
 __minimum_python_version__ = '3.6'
 __minimum_numpy_version__ = '1.16.0'
+__minimum_scipy_version__ = '0.18'
 # ASDF is an optional dependency, but this is the minimum version that is
 # compatible with Astropy when it is installed.
 __minimum_asdf_version__ = '2.5.2'

--- a/conftest.py
+++ b/conftest.py
@@ -3,9 +3,7 @@
 # This file is the main file used when running tests with pytest directly,
 # in particular if running e.g. ``pytest docs/``.
 
-from importlib.util import find_spec
 import os
-import pkg_resources
 import tempfile
 
 try:

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -91,6 +91,7 @@ templates_path.append('_templates')
 rst_epilog += """
 .. |minimum_python_version| replace:: {0.__minimum_python_version__}
 .. |minimum_numpy_version| replace:: {0.__minimum_numpy_version__}
+.. |minimum_scipy_version| replace:: {0.__minimum_scipy_version__}
 
 .. Astropy
 .. _`Astropy mailing list`: https://mail.python.org/mailman/listinfo/astropy

--- a/docs/install.rst
+++ b/docs/install.rst
@@ -15,7 +15,8 @@ Requirements
 
 ``astropy`` also depends on other packages for optional features:
 
-- `scipy`_: To power a variety of features in several modules.
+- `scipy`_ |minimum_scipy_version| or later: To power a variety of features
+  in several modules.
 
 - `h5py <http://www.h5py.org/>`_: To read/write
   :class:`~astropy.table.Table` objects from/to HDF5 files.

--- a/setup.cfg
+++ b/setup.cfg
@@ -53,7 +53,7 @@ test =
     coverage
     skyfield
 all =
-    scipy
+    scipy>=0.18
     h5py
     beautifulsoup4
     html5lib
@@ -105,6 +105,7 @@ filterwarnings =
     ignore:unclosed <ssl.SSLSocket:ResourceWarning
     ignore:The timefunc function is deprecated
     ignore:numpy.ufunc size changed:RuntimeWarning
+    ignore:Importing from numpy:DeprecationWarning:scipy
     ignore:Using or importing the ABCs from 'collections':DeprecationWarning
     ignore:can't resolve package from __spec__
     ignore:PY_SSIZE_T_CLEAN will be required for '#' formats

--- a/setup.cfg
+++ b/setup.cfg
@@ -106,6 +106,8 @@ filterwarnings =
     ignore:The timefunc function is deprecated
     ignore:numpy.ufunc size changed:RuntimeWarning
     ignore:Importing from numpy:DeprecationWarning:scipy
+    ignore:Conversion of the second argument:FutureWarning:scipy
+    ignore:Using a non-tuple sequence:FutureWarning:scipy
     ignore:Using or importing the ABCs from 'collections':DeprecationWarning
     ignore:can't resolve package from __spec__
     ignore:PY_SSIZE_T_CLEAN will be required for '#' formats

--- a/tox.ini
+++ b/tox.ini
@@ -65,6 +65,7 @@ deps =
     oldestdeps: numpy==1.16.*
     oldestdeps: matplotlib==2.1.*
     oldestdeps: asdf==2.4.*
+    oldestdeps: scipy==0.18.*
 
     # The devdeps factor is intended to be used to install the latest developer version
     # of key dependencies.

